### PR TITLE
Replace rebuild.sh / sourcesautoupdate.sh with a maintenance.sh to properly update cron job and auto-restart services when updating their code

### PR DESCRIPTION
**Pull request by alexAubin**
Follow up of https://github.com/YunoHost/apps/pull/2092

### CODE REVIEW

File: cron [hunk 1]
```diff
@@ -1,8 +1,14 @@
 # Every 4 hours
-0 */4 * * * root /bin/bash __BASEDIR__/rebuild.sh
+0 */4 * * * root /bin/bash __BASEDIR__/maintenance.sh rebuild_catalog
 
 # Everyday at 01:30 UTC
-30 1 * * * root /bin/bash __BASEDIR__/sourcesautoupdate.sh
+0 */4 * * * root /bin/bash __BASEDIR__/maintenance.sh autoupdate_app_sources
```
Comment (alexAubin) [inline cron:5]:
```suggestion
30 1* * * root /bin/bash __BASEDIR__/maintenance.sh autoupdate_app_sources
```

File: cron [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 
 # Every friday at 6 PM UTC
-0 17 * * 5 root /usr/bin/python3 __BASEDIR__/tools/update_app_levels/update_app_levels.py
+0 */4 * * * root /bin/bash __BASEDIR__/maintenance.sh update_app_levels
```
Comment (alexAubin) [inline cron:8]:
```suggestion
0 17 * * 5 root /bin/bash __BASEDIR__/maintenance.sh update_app_levels
```

File: cron [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 
 # Every friday at 6 PM UTC
-0 17 * * 5 root /usr/bin/python3 __BASEDIR__/tools/update_app_levels/update_app_levels.py
+0 17 * * 5 root /bin/bash __BASEDIR__/maintenance.sh update_app_levels
+
+# Every 6 hours
+0 */4 * * * root /bin/bash __BASEDIR__/maintenance.sh fetch_main_dashboard
+
+# Every day at 2AM
+0 */4 * * * root /bin/bash __BASEDIR__/maintenance.sh fetch_level_history
```
Comment (alexAubin) [inline cron:14]:
```suggestion
0 */6 * * * root /bin/bash __BASEDIR__/maintenance.sh fetch_main_dashboard

# Every day at 2AM
0 2 * * * root /bin/bash __BASEDIR__/maintenance.sh fetch_level_history
```

**Comment by Salamandar** [comment]
Yeah didnt read in detail but LGTM overall (famous last words 🙄).

Thanks for the refactoring, it looks really nice and having an unique scrit might pave the way to a smart systemd timer 😁

**Review by Salamandar — APPROVED** [review]

**Pull request merged**